src/vcu/src/controls.c

```diff
@@ -150,10 +219,62 @@ static int32_t get_regen_torque() {
     }
   }
 
-  // Regen is negative torque, and we've calculated a positive number so far
   return -1 * regen_torque;
 }
 
+static int32_t get_launch_control_speed(uint32_t front_wheel_speed) {
+uint32_t front_wheel_speedRPM = front_wheel_speed / 1000;
+  // Divide 100 because slip ratio is times 100, divide by 100 again because
+  // gear ratio is also multiplied by 100
+  int32_t target_speed = front_wheel_speedRPM * lc_settings.slip_ratio * LC_cGR / (100 * 100);
+  return target_speed;
+}
+
+static uint32_t get_front_wheel_speed() {
+  uint32_t left_front_speed;
+  uint32_t right_front_speed;
+
+  // If the 32 bit wheel speed is zero, then it's disconnect (in which case we should
+  // use the 16 bit one), or the wheels are actually not moving, so the 16 bit one will
+  // be zero as well
+  if (wheel_speeds.front_left_32b_wheel_speed == 0) {
+    left_front_speed = wheel_speeds.front_left_16b_wheel_speed;
+  } else {
+    left_front_speed = wheel_speeds.front_left_32b_wheel_speed;
+  }
+
+  if (wheel_speeds.front_right_32b_wheel_speed == 0) {
+    right_front_speed = wheel_speeds.front_right_16b_wheel_speed;
+  } else {
+    right_front_speed = wheel_speeds.front_right_32b_wheel_speed;
+  }
+
+  uint32_t avg_wheel_speed = left_front_speed/2 + right_front_speed/2;
+  return avg_wheel_speed;
+}
+
+static bool any_lc_faults() {
+  if (pedalbox_min(accel) < LC_ACCEL_BEGIN) {
+    printf("[LAUNCH CONTROL ERROR] Accel min (%d) too low\r\n", pedalbox_min(accel));
+    return true;
+  } else if (pedalbox.brake_2 > LC_BRAKE_BEGIN) {
```

Comment:
Why only brake_2? Don't we want the max of the 2 brakes?

Reply:
I actually don't remember the reason myself...should definitely go in a comment